source/game/system/CourseMap.hh

```diff
@@ -61,6 +61,25 @@ private:
     CourseMap();
     ~CourseMap() override;
 
+    void clearSectorChecked();
+    [[nodiscard]] s16 findSector(const EGG::Vector3f &pos, u16 checkpointIdx,
+            f32 &distanceRatio);
+    [[nodiscard]] s16 findSectorBetweenSides(const EGG::Vector3f &pos,
+            MapdataCheckPoint *checkpoint, f32 &distanceRatio);
+    [[nodiscard]] s16 findSectorOutsideSector(const EGG::Vector3f &pos,
+            MapdataCheckPoint *checkpoint, f32 &distanceRatio);
+    [[nodiscard]] s16 findSectorRegional(const EGG::Vector3f &pos, MapdataCheckPoint *checkpoint,
+            f32 &distanceRatio);
+    [[nodiscard]] s16 searchNextCheckpoint(const EGG::Vector3f &pos, s16 depth,
+            const MapdataCheckPoint *checkpoint, f32 &completion, bool playerIsForwards,
+            bool param_8) const;
+    [[nodiscard]] s16 searchPrevCheckpoint(const EGG::Vector3f &pos, s16 depth,
+            const MapdataCheckPoint *checkpoint, f32 &completion, bool playerIsForwards,
+            bool param_8) const;
+    [[nodiscard]] s16 findRecursiveSector(const EGG::Vector3f &pos, s16 depth,
+            bool searchBackwardsFirst, MapdataCheckPoint *checkpoint, f32 &completion,
+            bool playerIsForwards) const;
```

Comment:
- `findSector` is public. `findRecursiveSector` does not have to be, but it would look cleaner in a source file if it was. The two should be next to each other, as they are the most important functions.
- `clearSectorChecked` should be moved to the bottom, as it is the least important function.